operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/dependent/workflow/Condition.java

```diff
@@ -1,10 +1,11 @@
 package io.javaoperatorsdk.operator.processing.dependent.workflow;
 
+import java.util.Optional;
+
 import io.fabric8.kubernetes.api.model.HasMetadata;
 import io.javaoperatorsdk.operator.api.reconciler.Context;
-import io.javaoperatorsdk.operator.api.reconciler.dependent.DependentResource;
 
 public interface Condition<R, P extends HasMetadata> {
 
-  boolean isMet(DependentResource<R, P> dependentResource, P primary, Context<P> context);
+  boolean isMet(P primary, Optional<R> secondary, Context<P> context);
```

Comment:
Optionals should not be used as an argument. Can we just have it as a resource what is nullabel?

Reply:
While I generally agree, in this case, Optional conveys exactly the fact that the secondary resource might not exist (in the delete case) without needing additional documentation and calling this method is not under the user's control so there is no issue where a user could potentially pass `null` instead of an `Optional`… so I think it's better this way, though I guess we should document the method, either way :)

Reply:
So I would stick with standard Java convention, it feels quite weird, you could have this argument basically for any case. Having it just nullable, documented would be nice enough.